### PR TITLE
Add app's routes

### PR DESCRIPTION
**Pull request by oleh-momot**

### CODE REVIEW

File: registry/server/apps/services/prepareAppToInsert.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import _ from 'lodash/fp';
+
+import App, { AppBody } from '../interfaces';
+
+const prepareAppToInsert = (app: AppBody): App => {
+    const {
```
Comment (nightnei) [inline registry/server/apps/services/prepareAppToInsert.ts]: rewrite to 
```
const {
        dependencies = {},
        props = {},
        ssr,
        initProps = {},
    } = app;
```

File: registry/server/db.ts [hunk 2]
```diff
@@ -8,7 +8,7 @@ const client: string = config.get('database.client');
 const knexConf: knex.Config = { // after: const knex = require('knex')({client: 'mysql'});
     client: client,
     connection: config.get('database.connection'),
-
+    useNullAsDefault: true,
```
Comment (nightnei) [inline registry/server/db.ts]: it looks strange. please add a comment on why did you set it.

File: registry/server/apps/services/prepareAppToInsert.ts (hunk 1 above)
Comment (oleh-momot) [reply registry/server/apps/services/prepareAppToInsert.ts]: Updated

File: registry/server/db.ts (hunk 2 above)
Comment (oleh-momot) [reply registry/server/db.ts]: Added

File: registry/server/common/services/validateRequest.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+import _ from 'lodash/fp';
+
+const preProcessErrorResponse = _.compose<Array<Joi.ValidationError>, Array<Joi.ValidationErrorItem>, Array<string | undefined>, string>(
+    _.join('\n'),
+    _.map(_.get('message')),
+    _.get('details'),
+);
+
+type SelectDataToValidate = (req: Request) => any;
+export type ValidationPairs = Map<Joi.Schema, SelectDataToValidate>;
```
[the remark this answers is not in the document]
Comment (oleh-momot) [reply registry/server/common/services/validateRequest.ts]: Done

File: registry/server/apps/interfaces/index.ts [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+}
+
+export interface AppBody extends CommonApp {
+    dependencies?: AppDependencies,
+    props?: AppProps,
+    ssr: AppSSR,
+    initProps?: AppInitProps,
+}
+
+export default interface App extends CommonApp {
```
[the remark this answers is not in the document]
Comment (oleh-momot) [reply registry/server/apps/interfaces/index.ts]: Done

File: registry/server/apps/interfaces/index.ts [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+export type AppName = string;
+
+export interface CommonApp {
+    name: AppName,
+    spaBundle: string,
+    cssBundle: string,
+    assetsDiscoveryUrl?: string,
+}
+
+export interface AppBody extends CommonApp {
```
[the remark this answers is not in the document]
Comment (oleh-momot) [reply registry/server/apps/interfaces/index.ts]: Done

File: registry/server/apps/interfaces/index.ts [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+    src: string,
+    timeout: number,
+    primary: boolean,
+}
+
+export interface AppInitProps {
+    [propName: string]: any
+}
+
+export interface AppProps {
```
[the remark this answers is not in the document]
Comment (oleh-momot) [reply registry/server/apps/interfaces/index.ts]: Done

File: registry/server/apps/interfaces/index.ts [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+    [packageName: string]: string
+}
+
+export interface AppSSR {
+    src: string,
+    timeout: number,
+    primary: boolean,
+}
+
+export interface AppInitProps {
```
[the remark this answers is not in the document]
Comment (oleh-momot) [reply registry/server/apps/interfaces/index.ts]: Done

File: registry/server/apps/interfaces/index.ts [hunk 8]
```diff
@@ -0,0 +1,72 @@
+import Joi from '@hapi/joi';
+
+export interface AppDependencies {
+    [packageName: string]: string
+}
+
+export interface AppSSR {
```
[the remark this answers is not in the document]
Comment (oleh-momot) [reply registry/server/apps/interfaces/index.ts]: Done

File: registry/server/apps/interfaces/index.ts [hunk 9]
```diff
@@ -0,0 +1,72 @@
+import Joi from '@hapi/joi';
+
+export interface AppDependencies {
```
[the remark this answers is not in the document]
Comment (oleh-momot) [reply registry/server/apps/interfaces/index.ts]: Done

File: registry/server/apps/routes/deleteApp.ts [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+const validateRequestBeforeDeleteApp = validateRequest(new Map([
+    [Joi.object({
+        name: appNameSchema.required(),
+    }), selectParamsToValidate],
+]));
+
+const deleteApp = async (req: Request<DeleteAppRequestParams>, res: Response): Promise<void> => {
+    await validateRequestBeforeDeleteApp(req, res);
+
+    const {
```
[the remark this answers is not in the document]
Comment (oleh-momot) [reply registry/server/apps/routes/deleteApp.ts]: Done

File: registry/server/apps/interfaces/index.ts [hunk 11]
```diff
@@ -0,0 +1,72 @@
...
+
+export interface AppInitProps {
+    [propName: string]: any
+}
+
+export interface AppProps {
+    [propName: string]: any,
+}
+
+export type AppName = string;
```
[the remark this answers is not in the document]
Comment (oleh-momot) [reply registry/server/apps/interfaces/index.ts]: Done

File: registry/server/common/services/validateRequest.ts [hunk 12]
```diff
@@ -0,0 +1,36 @@
...
+
+export const selectQueryToValidate: SelectDataToValidate = _.get('query');
+export const selectParamsToValidate: SelectDataToValidate = _.get('params');
+export const selectBodyToValidate: SelectDataToValidate = _.get('body');
+
+const validateRequest = (validationPairs: ValidationPairs) => async (
```
[the remark this answers is not in the document]
Comment (oleh-momot) [reply registry/server/common/services/validateRequest.ts]: Done

File: registry/server/common/services/validateRequest.ts [hunk 13]
```diff
@@ -0,0 +1,36 @@
...
+const preProcessErrorResponse = _.compose<Array<Joi.ValidationError>, Array<Joi.ValidationErrorItem>, Array<string | undefined>, string>(
+    _.join('\n'),
+    _.map(_.get('message')),
+    _.get('details'),
+);
+
+type SelectDataToValidate = (req: Request) => any;
+export type ValidationPairs = Map<Joi.Schema, SelectDataToValidate>;
+
+export const selectQueryToValidate: SelectDataToValidate = _.get('query');
```
[the remark this answers is not in the document]
Comment (oleh-momot) [reply registry/server/common/services/validateRequest.ts]: Done

**Pull request merged**